app/coffee/DocumentManager.coffee

```diff
@@ -23,12 +23,14 @@ module.exports = DocumentManager =
 			return callback(error) if error?
 			if !lines? or !version?
 				logger.log {project_id, doc_id}, "doc not in redis so getting from persistence API"
-				PersistenceManager.getDoc project_id, doc_id, (error, lines, version, ranges, pathname, projectHistoryId) ->
+				PersistenceManager.getDoc project_id, doc_id, (error, lines, version, ranges, pathname, projectHistoryId, projectHistoryType) ->
 					return callback(error) if error?
-					logger.log {project_id, doc_id, lines, version, pathname, projectHistoryId}, "got doc from persistence API"
+					logger.log {project_id, doc_id, lines, version, pathname, projectHistoryId, projectHistoryType}, "got doc from persistence API"
 					RedisManager.putDocInMemory project_id, doc_id, lines, version, ranges, pathname, projectHistoryId, (error) ->
 						return callback(error) if error?
-						callback null, lines, version, ranges, pathname, projectHistoryId, null, false
+						RedisManager.setHistoryType doc_id, projectHistoryType, (error) ->
+							return callback(error) if error?
+							callback null, lines, version, ranges, pathname, projectHistoryId, null, false
```

Comment:
This is now bordering on the edge of what seems like too many callbacks. Think it's worth it to make this into a waterfall?

Reply:
Fair point, but I will leave it as this code will be removed again after we have turned off the old SL history.